File: filter-pattern/java/src/CriteriaMale.java
package src;

import java.util.*;

// 根据标准接口实现的过滤
public class CriteriaMale implements Criteria {

   @Override
   public List<Person> filter(List<Person> persons) {
      List<Person> malePersons = new ArrayList<Person>();
      for (Person person : persons) {
         if (person.getGender().equalsIgnoreCase("MALE")) {
            malePersons.add(person);
         }
      }
      return malePersons;
   }
}

File: filter-pattern/go/src/CriteriaMale.go
package src

import "strings"

// 根据标准接口实现的过滤
type CriteriaMale struct {
}

// 根据是否男性进行过滤
func (r *CriteriaMale) Filter(persons []*Person) []*Person {
  var malePersons []*Person
  for _, person := range persons {
    if strings.ToUpper(person.GetGender()) == "MALE" {
      malePersons = append(malePersons, person)
    }
  }
  return malePersons
}
